hetmech/hetmat/__init__.py

```diff
@@ -142,7 +144,12 @@ def read_first_matrix(specs):
             continue
         transpose = spec.get('transpose', False)
         file_format = spec.get('file_format', 'infer')
-        matrix = read_matrix(path, file_format=file_format)
+        try:
+            matrix = read_matrix(path, file_format=file_format)
+        except Exception as error:
+            logging.warning(f'Error reading matrix at {path}:\n{error}')
+            if delete_failures:
+                os.remove(path)
```

Comment:
We need a `continue` here. Also let's log the file deletion if it occurs.

Reply:
No need to import os because path is a pathlib.Path, so `path.unlink()` will work.